### PR TITLE
Dịch "model-construction_vn" - Phần 3

### LINKED ISSUE #1068: Dịch "model-construction_vn" - Phần 3
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_deep-learning-computation/model-construction_vn.md#L120-L217

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by tranyenthy**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #1068 

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_deep-learning-computation/model-construction_vn.md [hunk 1]
```diff
@@ -212,7 +228,10 @@ We can subclass `Block` to create layers (such as the `Dense` class provided by
 We exploit this versatility throughout the following chapters, especially when addressing convolutional neural networks.
 -->
 
-*dịch đoạn phía trên*
+Một ưu điểm quan trọng của trừu tượng hóa `Block` là tính linh hoạt của nó.
+Ta có thể phân lớp `Block` để tạo các tầng (chẳng hạn như lớp `Dense` được cung cấp bởi Gluon), toàn bộ các mô hình (như `MLP` ở phía trên) hoặc các thành phần đa dạng có độ phức tạp trung gian.
+Ta khai thác tính linh hoạt này xuyên suốt các chương sau, đặc biệt khi giải quyết các mạng nơ-ron tích chập.
```
Comment (minhduc0711) [inline chapter_deep-learning-computation/model-construction_vn.md:233]:
```suggestion
Ta sẽ tận dụng tính linh hoạt này xuyên suốt ở các chương sau, đặc biệt khi làm việc với các mạng nơ-ron tích chập. 
```

File: chapter_deep-learning-computation/model-construction_vn.md [hunk 2]
```diff
@@ -147,15 +148,24 @@ Typically this happens automatically.
 5. Initialize these parameters as needed.
 -->
 
-*dịch đoạn phía trên*
+1. Phương thức `forward` của nó chấp nhận đối số là dữ liệu đầu vào.
+2. Tạo ra một đầu ra bằng cách cho phương thức `forward` trả về một giá trị.
+Lưu ý rằng đầu ra có thể có kích thước khác với đầu vào.
+Ví dụ, tầng Dense đầu tiên trong mô hình phía trên nhận đầu vào có kích thước tùy ý nhưng trả về đầu ra có kích thước 256.
+3. Tính toán gradient của đầu ra theo đầu vào bằng phương thức `backward`.
+Thông thường điều này tự động xảy ra.
```
Comment (lkhphuc) [inline chapter_deep-learning-computation/model-construction_vn.md:156]:
```suggestion
3. Tính gradient của đầu ra theo đầu vào bằng phương thức `backward`, thường sẽ tự động xảy ra.
```

File: chapter_deep-learning-computation/model-construction_vn.md [hunk 3]
```diff
@@ -147,15 +148,24 @@ Typically this happens automatically.
...
 <!--
 In the following snippet, we code up a Block from scratch corresponding to a multilayer perceptron with one hidden layer with 256 hidden nodes, and a 10-dimensional output layer.
 Note that the `MLP` class below inherits the `Block` class.
 We will rely heavily on the parent class's methods, supplying only our own `__init__` and `forward` methods.
 -->
 
-*dịch đoạn phía trên*
+Trong đoạn mã dưới đây, chúng ta lập trình từ đầu một Block (Khối) tương đương với một perceptron đa tầng chỉ có một tầng ẩn và 256 nút ẩn, cùng một tầng đầu ra 10 chiều.
+Lưu ý rằng lớp `MLP` bên dưới đây kế thừa từ lớp `Block`.
+Ta sẽ phụ thuộc nhiều vào các phương thức của lớp cha, chỉ tự viết phương thức `__init__` và `forward`.
```
Comment (lkhphuc) [inline chapter_deep-learning-computation/model-construction_vn.md:168]: từ parent class trong lập trình bình thường mình cũng dịch là `lớp cha` hả mn?

File: chapter_deep-learning-computation/model-construction_vn.md [hunk 4]
```diff
@@ -186,7 +196,11 @@ To see why this is reasonable, imagine instantiating two MLPs, `net1` and `net2`
 Naturally, we would expect them them to represent two different learned models.
 -->
 
-*dịch đoạn phía trên*
+Để bắt đầu, ta sẽ tập trung vào phương thức `forward`.
+Lưu ý rằng nó nhận giá trị đầu vào `x`, tính toán biểu diễn ẩn (`self.hidden(x)`), và trả về giá trị đầu ra là các logit (`self.output( ... )`).
```
Comment (lkhphuc) [inline chapter_deep-learning-computation/model-construction_vn.md:200]:
```suggestion
Lưu ý rằng nó nhận giá trị đầu vào `x`, tính toán tầng biểu diễn ẩn (`self.hidden(x)`) và trả về các giá trị logit (`self.output( ... )`).
```

File: chapter_deep-learning-computation/model-construction_vn.md [hunk 5]
```diff
@@ -127,14 +127,15 @@ This is actually just shorthand for `net.forward(X)`, a slick Python trick achie
 ## A Custom Block
 -->
 
-## *dịch tiêu đề phía trên*
+## Một Block Tùy chỉnh
```
Comment (lkhphuc) [inline chapter_deep-learning-computation/model-construction_vn.md:130]:
```suggestion
## Một Khối Tùy chỉnh
```
Chỗ nào nhắc đến `block` trong code mình mới để nguyên.

File: chapter_deep-learning-computation/model-construction_vn.md (hunk 3 above)
Comment (minhduc0711) [reply chapter_deep-learning-computation/model-construction_vn.md:168]: mình cũng không rõ, search thử thì thấy có vẻ cũng đc dùng nhiều

Comment (duythanhvn) [reply chapter_deep-learning-computation/model-construction_vn.md:168]: Mình cũng thấy người ta hay dùng lớp con (sub-class) và lớp cha (super-class, parent-class). Cá nhân mình thường chỉ dùng lớp con, lớp kế thừa (sub-class) và lớp được kế thừa (parent-class) hay lớp cơ sở (base-class). (Mình nghe "con-tra" thấy mệt quá).

Mà cái này tuỳ ngôn ngữ mà người ta gọi, như Java mình hay nghe lớp con-lớp cha, nhưng trong C++ thì thầy mình lại dùng lớp cơ sở và lớp kế thừa. Hay là nó tuỳ vào độ "cổ" của ngôn ngữ lập trình? :v

Comment (lkhphuc) [reply chapter_deep-learning-computation/model-construction_vn.md:168]: Nếu muốn politically correct thì mình phải dịch là `lớp phụ huynh`. :))

Comment (duythanhvn) [reply chapter_deep-learning-computation/model-construction_vn.md:168]: > Nếu muốn politically correct thì mình phải dịch là `lớp phụ huynh`. :))

Dịch "trần trụi" ghê :))))
